src/ray/core_worker/transport/direct_actor_task_submitter.cc

```diff
@@ -485,6 +501,18 @@ void CoreWorkerDirectActorTaskSubmitter::PushActorTask(ClientQueue &queue,
         reply_callback(status, reply);
       };
 
+  if (force_fail) {
```

Comment:
Do we have to go through the PushActorTask logic here? Why not just do it in SendPendingTasks when we first assign force_fail? That seems like it would produce fewer code changes.

Reply:
Code updated. I moved the `if` to `SendPendingTasks`. Due to the fact that we need to reuse the reply callback for failure handling, I created a new private method `ReplyCallback` to hold the callback logic.